olewoo_interop/typestring.cpp

```diff
@@ -74,17 +74,17 @@ void stringifyTypeDesc(TYPEDESC* typeDesc, ITypeInfo* pTypeInfo, IDLFormatter_io
         // VARIANT/VARIANTARG compatible types
     case VT_I2: ift->AddString( "short"); return;
     case VT_I4: ift->AddString( "long"); return;
-    case VT_R4: ift->AddString( "float"); return;
+    case VT_R4: ift->AddString( "single"); return;
     case VT_R8: ift->AddString( "double"); return;
-    case VT_CY: ift->AddString( "CY"); return;
+    case VT_CY: ift->AddString( "CURRENCY"); return;
     case VT_DATE: ift->AddString( "DATE"); return;
     case VT_BSTR: ift->AddString( "BSTR"); return;
     case VT_DISPATCH: ift->AddString( "IDispatch*"); return;
     case VT_ERROR: ift->AddString( "SCODE"); return;
     case VT_BOOL: ift->AddString( "VARIANT_BOOL"); return;
     case VT_VARIANT: ift->AddString( "VARIANT"); return;
     case VT_UNKNOWN: ift->AddString( "IUnknown*"); return;
-    case VT_UI1: ift->AddString( "BYTE"); return;
+    case VT_UI1: ift->AddString( "unsigned char"); return;
```

Comment:
A bit unsure about this.
```
 typedef byte BYTE;
```
[Source](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-oaut/25955917-9c7b-4153-9dd4-d229d77c3ede)

Reply:
I know - it looks odd - however I'm going by the output that oleview generates.  
(try it on the dll attached to the PR!)

Reply:
Hmm. Now I'm doubting the specifications because `byte` isn't a C/C++ data type. 

From [Windows Data Types](https://docs.microsoft.com/en-us/windows/win32/winprog/windows-data-types#BYTE):
```
A byte (8 bits).
This type is declared in WinDef.h as follows:
typedef unsigned char BYTE;
```

I don't see any evidence of `byte` being a C/C++ native data type. Therefore, I think `typedef byte BYTE;` is invalid, so I guess we're in agreement now.